Update to doctrine/coding-standard 10 on 4.0.x

greg0ire opened the pull request:
To be followed by a `git merge origin/3.5.x --strategy ours`

greg0ire committed:
Manually fix issues with lines that are too long.

greg0ire committed:
Ignore false positive of coding standard sniff

morozov commented on src/Cache/QueryCacheProfile.php:
```diff
@@ -22,12 +22,12 @@ class QueryCacheProfile
         private readonly int $lifetime = 0,
-        private readonly ?string $cacheKey = null,
-        private readonly ?CacheItemPoolInterface $resultCache = null
+        private readonly string|null $cacheKey = null,
+        private readonly CacheItemPoolInterface|null $resultCache = null,
```
I don't believe that union types should be enforced over the nullable ones. We discussed it recently in another thread.

greg0ire replied:
Can you please open an issue on the CS repo? I think that's were the issue needs to be discussed.

morozov replied:
Done: https://github.com/doctrine/coding-standard/issues/288.

greg0ire commented:
Psalm suddenly chokes on `Connection&MockObject`

Psalm 5.0.0-beta1 doesn't have the issue but comes with other errors…

greg0ire committed:
Avoid intersection types

They are not supported by Psalm yet.

morozov reviewed (changes requested):
See https://github.com/doctrine/dbal/pull/5632#discussion_r957742275.

greg0ire commented:
I'll revert the change about union types that way we don't have to wait for the resolution of https://github.com/doctrine/coding-standard/issues/288 to  merge this.

greg0ire committed:
Upgrade to doctrine/coding-standard 10

One rule is excluded temporarily, pending a discussion on the standard
repository.
